pkg/jobrunaggregator/jobtableprimer/job_name_generator.go

```diff
@@ -0,0 +1,263 @@
+		for _, periodicURL := range s.periodicURLs {
+			if strings.Contains(periodicURL, release.Release) {
+				found = true
+				break
+			}
+		}
+		for _, releaseConfigURL := range s.releaseConfigURLs {
+			if strings.Contains(releaseConfigURL, release.Release) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			newReleases = append(newReleases, release.Release)
+		}
+	}
+	for _, release := range newReleases {
+		s.addReleaseURLs(release)
+	}
+}
+
+func (s *jobNameGenerator) GenerateJobNames() ([]string, error) {
+	jobNames := []string{}
+
+	for _, url := range s.releaseConfigURLs {
+		resp, err := http.Get(url)
+		if err != nil {
+			return jobNames, fmt.Errorf("error reading %v: %w", url, err)
+		}
+		if resp.StatusCode < 200 || resp.StatusCode > 299 {
+			return jobNames, fmt.Errorf("error reading %v: %v", url, resp.StatusCode)
+		}
+
+		content, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return jobNames, fmt.Errorf("error reading %v: %w", url, err)
+		}
+		resp.Body.Close()
+
+		releaseConfig := &FakeReleaseConfig{}
+		if err := json.Unmarshal(content, releaseConfig); err != nil {
+			return jobNames, fmt.Errorf("error reading %v: %w", url, err)
+		}
+
+		jobNames = append(jobNames, fmt.Sprintf("// begin %v", url))
+		localLines := []string{}
+		for _, curr := range releaseConfig.Verify {
+			localLines = append(localLines, curr.ProwJob.Name)
+		}
+		sort.Strings(localLines)
+		jobNames = append(jobNames, localLines...)
+		jobNames = append(jobNames, fmt.Sprintf("// end %v", url))
+		jobNames = append(jobNames, "")
+	}
+
+	for _, url := range s.periodicURLs {
+		resp, err := http.Get(url)
+		if err != nil {
+			return jobNames, fmt.Errorf("error reading %v: %w", url, err)
+		}
+		if resp.StatusCode < 200 || resp.StatusCode > 299 {
+			return jobNames, fmt.Errorf("error reading %v: %v", url, resp.StatusCode)
+		}
+
+		content, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return jobNames, fmt.Errorf("error reading %v: %w", url, err)
+		}
+		resp.Body.Close()
+
+		periodicConfig := &FakePeriodicConfig{}
+		if err := yaml.Unmarshal(content, periodicConfig); err != nil {
+			return jobNames, fmt.Errorf("error reading %v: %w", url, err)
+		}
+
+		jobNames = append(jobNames, fmt.Sprintf("// begin %v", url))
+		localLines := []string{}
+		for _, curr := range periodicConfig.Periodics {
+			// TODO: the single file for say "master" actually contains every release, but we only want jobs 4.10+
+			// where we started disruption monitoring. Adding a bunch of future rows to buy us time but this could
+			// stand some logic.
+			if len(s.releases) > 0 {
+				foundRelease := false
+				for _, release := range s.releases {
+					if release.Major > 4 || (release.Major == 4 && release.Minor > 9) {
+						if strings.Contains(curr.Name, "-"+release.Release) {
+							foundRelease = true
+							break
+						}
+					}
+				}
+				if !foundRelease {
+					continue
+				}
+			} else if !strings.Contains(curr.Name, "-4.10") &&
```

Comment:
Is this an expected case where we wouldn't have any releases at this point?  Just wondering if it is in effect dead code that we don't expect to run and may as well clean up?

Reply:
These are kept here for ./job-run-aggregator generate-job-names command to continue working. Most likely there is no other users other than us. But I am just trying to be cautious. When we decide to remove the command, we will remove the job in the continuous-release-jobs repo and these code sections from ci-tools as well.  This applies to the two comments above as well.

Reply:
Oh, interesting so you only call

jobNameGenerator.UpdateURLsForNewReleases(releases) for [createJobRowsFromReleases](https://github.com/xueqzhan/ci-tools/blob/2ccf98edeb22509ee82f535d6c70861f3b18a528/pkg/jobrunaggregator/jobtableprimer/create_jobs.go#L18) but not for [NewGenerateJobNamesCommand](https://github.com/xueqzhan/ci-tools/blob/2ccf98edeb22509ee82f535d6c70861f3b18a528/pkg/jobrunaggregator/jobtableprimer/generate_jobnames.go#L28).  Don't we need it there too?

Reply:
Or is this a two step process and the expectation is that  generate-job-names  is obsoleted and removed later?

Reply:
generate-job-names will be removed, hopefully soon. It uses the default config as it is. So it is not calling jobNameGenerator.UpdateURLsForNewReleases(releases). There is no dependency on bigquery for that command to work. It is better to keep it that way. It will be removed later anyhow.